178 add benchmarks for custom pallets per runtime

bogdanS98 opened the pull request:
Closes #178.

ebma commented:
> Note:
>- parachain-staking and issue pallets benchmarks are not yet included in any runtime.
>- nomination , redeem , replace, vault-registry pallets benchmarks are not yet included in Pendulum runtime.

Should we just add them then or is there a good reason not to?

> Please ensure that the collator nodes are redeployed after this PR is merged.

What changes require the redeployment of the collator nodes?

bogdanS98 commented:
> Should we just add them then or is there a good reason not to?
> 
> >

Yes, we should add them. They're not added yet because I'm still getting some errors on some benchmarks and I'm trying to fix them. 
Meanwhile everything that's already added in the PR can be reviewed as it's independent of the missing pallets.

> What changes require the redeployment of the collator nodes?
> 
> >

I modified `Cargo.toml` and `chain_spec.rs` files from the `node` directory;  updated Spacewalk pallets versions and some minor changes in the genesis configs.

ebma commented on runtime/foucoco/src/lib.rs:
```diff
@@ -1252,13 +1252,22 @@ impl currency::CurrencyConversion<currency::Amount<Runtime>, CurrencyId> for Cur
 
 parameter_types! {
 	pub const RelayChainCurrencyId: CurrencyId = XCM(0);
+	pub const GetWrappedCurrencyId: CurrencyId = CurrencyId::Stellar(Asset::AlphaNum4 {
```
Should we add a comment that describes the asset? Like, why are we using this one.

ebma commented on runtime/pendulum/src/lib.rs:
```diff
@@ -1178,13 +1178,22 @@ impl currency::CurrencyConversion<currency::Amount<Runtime>, CurrencyId> for Cur
 }
 parameter_types! {
 	pub const RelayChainCurrencyId: CurrencyId = XCM(0); // 0 is the index of the relay chain in our XCM mapping
+	pub const GetWrappedCurrencyId: CurrencyId = CurrencyId::Stellar(Asset::AlphaNum4 {
```
Also here, let's say what this is.

ebma commented on node/src/chain_spec.rs:
```diff
@@ -645,7 +645,7 @@ fn foucoco_genesis(
 		},
 		vault_registry: foucoco_runtime::VaultRegistryConfig {
-			minimum_collateral_vault: vec![(XCM(0), 3_000_000_000_000)],
+			minimum_collateral_vault: vec![(XCM(0), 0)],
```
Why `0`? I guess a higher amount would also work?

ebma commented on runtime/amplitude/Cargo.toml:
```diff
@@ -274,8 +275,9 @@ runtime-benchmarks = [
-    "orml-currencies-allowance-extension/runtime-benchmarks",
-    "treasury-buyout-extension/runtime-benchmarks",
+    "parachain-staking/runtime-benchmarks",
+	"orml-currencies-allowance-extension/runtime-benchmarks",
+	"treasury-buyout-extension/runtime-benchmarks"
```
Nit: indent

ebma commented on runtime/amplitude/src/lib.rs:
```diff
@@ -1187,13 +1187,22 @@ impl currency::CurrencyConversion<currency::Amount<Runtime>, CurrencyId> for Cur
 }
 parameter_types! {
 	pub const RelayChainCurrencyId: CurrencyId = XCM(0); // 0 is the index of the relay chain in our XCM mapping
+	pub const GetWrappedCurrencyId: CurrencyId = CurrencyId::Stellar(Asset::AlphaNum4 {
```
Nit: Comment

ebma reviewed (commented):
Looks good to me so far 👍

ebma commented:
Is this now ready for a re-review or are you still adding more changes to it?

bogdanS98 commented:
> Is this now ready for a re-review or are you still adding more changes to it?

Still adding changes

ebma commented:
@bogdanS98 what's the status with this? Do you need help here?

vadaynujra commented:
Any actions for us to take wrt the node upgrades @ebma ?

ebma commented:
Good question @vadaynujra. @bogdanS98 in the description you mention that the nodes need to be redeployed. Looking at the PR, there are some changes to files in the node folder. However, to me it seems like these changes don't really require a redeployment. Changing the chain spec does not affect the operation of the nodes. What do you think?

bogdanS98 commented:
@ebma Indeed changing the chain spec does not affect the operation of the nodes, I'll update the description 👍🏼

bogdanS98 commented on pallets/parachain-staking/src/benchmarking.rs:
```diff
@@ -145,13 +145,12 @@ benchmarks! {
-		let max_col_reward = InflationConfig::<T>::get().collator.reward_rate.per_block * MaxCollatorCandidateStake::<T>::get() * MaxSelectedCandidates::<T>::get().into();
+		let max_col_reward = InflationConfig::<T>::get().collator.reward_rate.per_block * MaxCollatorCandidateStake::<T>::get() * (MaxSelectedCandidates::<T>::get() as u64).into();
 		let network_block_reward = T::NetworkRewardRate::get() * max_col_reward;
-		assert!(new_issuance > issuance);
```
This line was removed since our `NetworkRewardRate` is set to 0 (e.g. [here](https://github.com/pendulum-chain/pendulum/blob/991c7694084c22ee61716723d001163e746a1725/runtime/pendulum/src/lib.rs#L916))

ebma replied:
Instead of removing the check completely, maybe it would be better to use an if condition only 
```
if network_block_reward > 0 {
   assert!(new_issuance > issuance)
}
```

ebma commented on README.md:
```diff
@@ -162,7 +163,9 @@ runtimes `lib.rs` file.
-    --output runtime/pendulum/src/weights/
+    --output runtime/pendulum/src/weights/ \
+    --template .maintain/frame-weight-template.hbs
+
```
Nit: remove extra new line

ebma commented on node/src/chain_spec.rs:
```diff
@@ -604,11 +608,13 @@ fn amplitude_genesis(
-			)],
-			batching_api: b"http://dia-00.pendulumchain.tech:8070/currencies".to_vec(),
+			supported_currencies: vec![
+				foucoco_runtime::AssetId::new(b"Amplitude".to_vec(), b"AMPE".to_vec()),
+				foucoco_runtime::AssetId::new(b"Kusama".to_vec(), b"KSM".to_vec()),
+				foucoco_runtime::AssetId::new(b"Stellar".to_vec(), b"XLM".to_vec()),
+				foucoco_runtime::AssetId::new(b"FIAT".to_vec(), b"USD-USD".to_vec()),
```
Nit: I messed this up, let's use the proper imports
```suggestion
				amplitude_runtime::AssetId::new(b"Amplitude".to_vec(), b"AMPE".to_vec()),
				amplitude_runtime::AssetId::new(b"Kusama".to_vec(), b"KSM".to_vec()),
				ampltiude_runtime::AssetId::new(b"Stellar".to_vec(), b"XLM".to_vec()),
				amplitude_runtime::AssetId::new(b"FIAT".to_vec(), b"USD-USD".to_vec()),
```

ebma commented on node/src/chain_spec.rs:
```diff
@@ -565,30 +568,31 @@ fn amplitude_genesis(
+				Key::ExchangeRate(CurrencyId::Native),
 				Key::ExchangeRate(CurrencyId::XCM(0)),
-				Key::ExchangeRate(MAINNET_USDC_CURRENCY_ID),
+				Key::ExchangeRate(TESTNET_USDC_CURRENCY_ID),
```
Nit: I changed this for testing and forgot to change it back
```suggestion
				Key::ExchangeRate(MAINNET_USDC_CURRENCY_ID),
```

ebma commented on node/src/chain_spec.rs:
```diff
@@ -1013,5 +973,16 @@ fn pendulum_genesis(
+		dia_oracle_module: pendulum_runtime::DiaOracleModuleConfig {
+			authorized_accounts: authorized_oracles,
+			supported_currencies: vec![
+				foucoco_runtime::AssetId::new(b"Pendulum".to_vec(), b"PEN".to_vec()),
+				foucoco_runtime::AssetId::new(b"Polkadot".to_vec(), b"DOT".to_vec()),
+				foucoco_runtime::AssetId::new(b"Stellar".to_vec(), b"XLM".to_vec()),
+				foucoco_runtime::AssetId::new(b"FIAT".to_vec(), b"USD-USD".to_vec()),
```
```suggestion
				pendulum_runtime::AssetId::new(b"Pendulum".to_vec(), b"PEN".to_vec()),
				pendulum_runtime::AssetId::new(b"Polkadot".to_vec(), b"DOT".to_vec()),
				pendulum_runtime::AssetId::new(b"Stellar".to_vec(), b"XLM".to_vec()),
				pendulum_runtime::AssetId::new(b"FIAT".to_vec(), b"USD-USD".to_vec()),
```

ebma commented on runtime/amplitude/src/lib.rs:
```diff
@@ -1223,7 +1233,8 @@ impl stellar_relay::Config for Runtime {
 	type ValidatorLimit = ValidatorLimit;
 	type IsPublicNetwork = IsPublicNetwork;
-	type WeightInfo = stellar_relay::SubstrateWeight<Runtime>;
+	//TODO change this
```
Nit: let's remove the comment

ebma commented on runtime/foucoco/src/lib.rs:
```diff
@@ -1257,7 +1266,8 @@ impl stellar_relay::Config for Runtime {
 	type ValidatorLimit = ValidatorLimit;
 	type IsPublicNetwork = IsPublicNetwork;
-	type WeightInfo = stellar_relay::SubstrateWeight<Runtime>;
+	//TODO change this
```
Also here

ebma reviewed (approved):
LGTM
